.github/workflows/ci.yml

```diff
@@ -0,0 +1,103 @@
+name: Pipeline
+on: [push]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Start timer
+        id: start_timer
+        run: |
+         echo "START_TIME=$(date +%s)" >> $GITHUB_ENV
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '18'
+      - name: Install dependencies
+        run: cd angular/ && npm install
+      - name: Run tests
+        run: cd angular/ && npm run coverage
```

Comment:
let'sd setup setup working directory for your job. instead of using `cd` cli in steps.